### PR TITLE
Move sockets includes to a sockets directory

### PR DESCRIPTION
**Pull request by bremoran**
Leave include forwarders in place in old directory.

### CODE REVIEW

File: mbed-net-sockets/Socket.h [hunk 1]
```diff
@@ -1,32 +1,22 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
 #ifndef __MBED_NET_SOCKETS_SOCKET_H__
 #define __MBED_NET_SOCKETS_SOCKET_H__
-
-/* Include all versions of the API */
-#include "v0/Socket.h"
```
Comment (bogdanm) [inline mbed-net-sockets/Socket.h]: If you remove this, doesn't the change become breaking?

Comment (bremoran) [inline mbed-net-sockets/Socket.h]: It's just moved to another place.  Look at the commits in the PR in order and it makes more sense.

**Comment by bremoran** [comment]
In order to review this PR *please* look at the commits in order, here: https://github.com/ARMmbed/sockets/pull/31/commits

This change is non-breaking, but will generate lots of warnings.

cc @bogdanm @hugovincent @autopulated @0xc0170 @sg- @yogpan01 @SeppoTakalo

**Comment by bremoran** [comment]
(jenkins is happy now)

**Comment by SeppoTakalo** [comment]
CC: @artokin

**Comment by autopulated** [comment]
:+1:

Also: 

:fireworks:  :sparkling_heart: :rainbow: :sparkles: non-breaking change! :sparkles: :rainbow: :sparkling_heart: :fireworks:

**Comment by bogdanm** [comment]
What, no unicorns?
+1 :smile:

**Comment by autopulated** [comment]
Unicorn emoji comes in unicode v8 :sob: I added a custom one to slack though ;)

**Comment by 0xc0170** [comment]
+1

**Pull request merged**